lib/Myriad/Service/Implementation.pm

```diff
@@ -317,7 +317,16 @@ async method load () {
                     });
                     await $self->rpc->reply_success($service_name, $message, $response);
                 } catch ($e) {
-                    $e = Myriad::Exception::InternalError->new(reason => $e) unless (blessed $e and $e->does('Myriad::Exception'));
+                    # Pass through the myriad exception, use indirect exports from Myriad::Exception::Base
+                    # for simplicity
+                    unless ((ref $e) =~ /^Myriad::Exception::/) {
+                        # Ensure that we are passing the exception reason only by means of scalar
+                        # or it would cause issue with strongly typed clients.
+                        # Moreover, attempt to use blessed object would crash the listener
+                        $e = "$e" if ref $e;
+                        $e =~ s/(\s+)$//; # Drop the trailling spaces
+                        $e = Myriad::Exception::InternalError->new(reason => $e);
```

Comment:
The `reason` handling in the exception class itself should be responsible for ensuring that `reason` is a valid string, converting as necessary - not the caller.